packages/augur-sdk/src/state/db/ZeroXOrders.ts

```diff
@@ -120,20 +120,24 @@ export class ZeroXOrders extends AbstractTable {
     this.cashAssetData = `0xf47261b0000000000000000000000000${cashTokenAddress}`;
     this.shareAssetData = `0xa7cb5fb7000000000000000000000000${shareTokenAddress}000000000000000000000000000000000000000000000000000000000000008000000000000000000000000000000000000000000000000000000000000000a000000000000000000000000000000000000000000000000000000000000000c0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000`;
     this.takerAssetData = `0xa7cb5fb7000000000000000000000000${this.tradeTokenAddress}000000000000000000000000000000000000000000000000000000000000008000000000000000000000000000000000000000000000000000000000000000a000000000000000000000000000000000000000000000000000000000000000c0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000`;
+
+    this.subscribeToOrderEvents();
   }
 
   static async create(db: DB, networkId: number, augur: Augur): Promise<ZeroXOrders> {
     const zeroXOrders = new ZeroXOrders(db, networkId, augur);
     await zeroXOrders.clearDB();
-    await zeroXOrders.subscribeToMeshEvents();
     return zeroXOrders;
   }
 
-  async subscribeToMeshEvents(): Promise<void> {
-    return this.augur.zeroX.subscribeToMeshEvents(this.handleMeshEvent.bind(this));
+  subscribeToOrderEvents() {
+    // This only works if `zeroX` has been set on the augur instance and
+    // it doesn't get over-written so... something.
```

Comment:
Should `zeroX` be hidden behind getter-setter interfaces so overwrite can be handled?

Reply:
The getter/setter interface doesn't work in this direction.

the `ZeroX` instance is a server-side class, an it basically subscribes to the the mesh events of the things it owns. On the server-side this is the rpcClient. I need to add another layer to proxy events from the Client side to the Server side over the connector for the BrowserMesh.

Once that's the case the `mesh` instance on `ZeroX` should be a wrapper object which actually listens to global events an then re-emits them. Or that's my thinking at a least.

Currently, when just implemented with the SingleThreadConnector it's using a real BrowserMesh instance as the event source.